docker-plugin/src/main/resources/com/nirima/jenkins/plugins/docker/DockerJobProperty/config.jelly

```diff
@@ -2,21 +2,22 @@
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
 
     <f:optionalBlock title="Docker Container" name="hasDockerContainer" inline="true" checked="${instance!=null}" >
-        <f:entry title="${%Commit on successful build}" field="tagOnCompletion">
-            <f:checkbox default="false"/>
-        </f:entry>
+        <f:optionalBlock title="${%Commit on successful build}" field="tagOnCompletion" inline="true">
```

Comment:
JobProperty was put into optionalBlock with inline=true, because jenkins core sucks and doesn't wrap it in existed UI (i have idea how fix it, but it need some time). In your case you will be forced to wrap all this options in if statements in java code - that usually sucks by design.